pkg/costmodel/cluster_helpers.go

```diff
@@ -640,7 +640,12 @@ func buildLabelsMap(
 		// ingested label data. This removes the label_ prefix that prometheus
 		// adds to emitted labels. It also keeps from ingesting prometheus labels
 		// that aren't a part of the asset.
-		m[key] = result.GetLabels()
+		if _, ok := m[key]; !ok {
+			m[key] = map[string]string{}
+		}
+		for k, l := range result.GetLabels() {
+			m[key][k] = l
```

Comment:
This is _way_ better, but will still result in one value for a `key` "winning" in the event of a conflict. It's not really possible to avoid that. I'm wondering if, in the future, there is a way for us to preserve conflicts instead of picking one winner silently. Not for this PR.

Reply:
Right, good catch! If there are two `kube_node_labels` metrics describing the same node and using the same labels, the second label read wins (example below). I can't think of a simple solution right now to resolve those conflicts, but we may be able to implement something similar to the `honor_labels` Prometheus config ([docs ref](https://prometheus.io/docs/prometheus/latest/configuration/configuration/)).

```
kube_node_labels{node="gke-qa-gcp1-default-pool-169fb9bb-0zz6", label_topology_gke_io_zone="us-east1-d"}
kube_node_labels{node="gke-qa-gcp1-default-pool-169fb9bb-0zz6", label_topology_gke_io_zone="us-east1-a"}
```